astropy/coordinates/angles.py

```diff
@@ -644,12 +650,20 @@ def _wrap_internal(self):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
+        # TODO: This could probably be sped up as well.  u.degree.to(self.unit)
```

Comment:
True enough, but is it worth adding the complexity? Also, thinking more, I mostly dislike it here in the code, which is becoming messy (I would've liked to remove the `if` statement below too if it didn't have such a very obvious perfermance benefti). I'd be less uncomfortable if we precalculated this in `__new__` (though that needs some thought to ensure that `__array_finalize__` still does the right thing for views, etc.).